lib/fli/FliImpl.h

```diff
@@ -227,14 +346,45 @@ class FliTimerCache {
     FliImpl *impl;
 };
 
+class FliIterator : public GpiIterator {
+public:
+    enum OneToMany {
+        OTM_END = 0,
+        OTM_CONSTANTS,  // include Generics
+        OTM_SIGNALS,
+        OTM_REGIONS,
+        OTM_SIGNAL_SUB_ELEMENTS,
+        OTM_VARIABLE_SUB_ELEMENTS
+    };
+
+    FliIterator(GpiImplInterface *impl, GpiObjHdl *hdl);
+
+    virtual ~FliIterator() { };
+
+    Status next_handle(std::string &name, GpiObjHdl **hdl, void **raw_hdl);
+
+private:
+    void populate_handle_list(OneToMany childType);
+
+private:
+    static GpiIteratorMapping<int, OneToMany> iterate_over;      /* Possible mappings */
+    std::vector<OneToMany> *selected;                            /* Mapping currently in use */
+    std::vector<OneToMany>::iterator one2many;
+
+    std::vector<void *> m_vars;
```

Comment:
@stuarthodgson 
I originally used the actual types, but the issue is each one (m_vars, m_sigs, and m_regs) will each be a unique type since the FLI interface likes to have a different handle type for the different kinds and that would then require a different "m_iterator" for each specific type.  This was the cleanest way to not have to test in the next_handle() function which iterator to use.